### ISSUE TITLE
Enable async execution if result is a Promise

### ISSUE THREAD

**Issue by n-riesco** [n-riesco/nel#6]
Moved from https://github.com/n-riesco/jp-babel/issues/17

Examples of usage:

1. When a cell execution returns a promise; e.g.

    ```js
    getPromise();
    ```
   this would be equivalent to

    ```js
    (function($$) {
        $$.async();

        getPromise().then(function(result) {
            $$.sendResult(result);
        }).catch(function(error) {
            $$.sendError(error);
        });
    }($$);
    ```

2. Similarly with other output methods, e.g.:

    ```js
    $$.html(getPromise());
    ```

   this would be equivalent to:

    ```js
    (function($$) {
        $$.async();

        getPromise().then(function(result) {
            $$.html(result);
        }).catch(function(error) {
            $$.sendError(error);
        });
    }($$);
    ```

**Comment by thalesmello** [comment]
Would it work with something like

```js
getPromise().then($$.html);
```

?

**Comment by n-riesco** [comment]
@thalesmello (haven't tested it) but yes, it should.

**Comment by n-riesco** [comment]
And more concisely:

```js
(function($$) {
    $$.async();
    getPromise().then($$.html, $$.sendError);
}($$);

```

**Comment by n-riesco** [comment]
To be clear, it should work because functions like `$$.html` are bound functions (i.e. when `$$` is created, the value of `this` in functions like `$$.html` is bound to `$$`).

**Comment by thalesmello** [comment]
@n-riesco I've created a candidate implementation in #8 

Would you be kind to take a look?

**Comment by n-riesco** [comment]
Closed by 7cda5be4e86287f0ed6b59a691337a9f92f22511
